pydash/pydash_web/controller/login.py

```diff
@@ -0,0 +1,40 @@
+"""
+Manages the logging in of a user into the application,
+and rejecting visitors that enter improper sign-in information.
+"""
+from flask import render_template, flash, redirect, url_for, request
+from flask_login import current_user, login_user, logout_user, login_required
+import pydash_app.user
+
+from pydash_web.forms import LoginForm
+
+
+def login():
+    if current_user.is_authenticated:
+        return redirect(url_for("dashboard"))
+
+    login_form = LoginForm()
+    if not login_form.validate_on_submit():
+        return __render_login_form(login_form)
+
+    user = pydash_app.user.authenticate(login_form.username.data,
+                                        login_form.password.data)
+    if not user:
+        flash("Wrong username or password!")
+        return __render_login_form(login_form)
+
+    login_user(user, remember=login_form.remember_me.data)
+    flash("User {} successfully logged in!".format(user.name))
+    return redirect(__next_page())
+
+
+def __next_page():
+    next = request.args.get('next')
+    if not next or url_parse.netloc(next) != '':
```

Comment:
undefined url_parse

Reply:
Import has been added. I have no idea why even without the import, Flask was able to run this script without any problems. :man_shrugging: